### PR TITLE
cli: Replace linters with ruff, upgrade mypy, upgrade black, run autofixes

### PR DESCRIPTION
**Pull request by underyx**
PR checklist:

- [x] Purpose of the code is [evident to future readers](https://semgrep.dev/docs/contributing/contributing-code/#explaining-code)
- [x] Tests included or PR comment includes a reproducible test plan
- [x] Documentation is up-to-date
- [x] A changelog entry was [added to changelog.d](https://semgrep.dev/docs/contributing/contributing-code/#adding-a-changelog-entry) for any user-facing change
- [x] Change has no security implications (otherwise, ping security team)

If you're unsure on any of this, please see:

- [Contribution guidelines](https://semgrep.dev/docs/contributing/contributing-code)!
- [One of the more specific guides located here](https://semgrep.dev/docs/contributing/contributing/)

### CODE REVIEW

File: cli/src/semgrep/autofix.py [hunk 1]
```diff
@@ -259,4 +255,4 @@ def apply_fixes(rule_matches_by_rule: RuleMatchMap, dryrun: bool = False) -> Non
                 f"successfully modified {unit_str(len(modified_files), 'file')}."
             )
         else:
-            logger.info(f"no files modified.")
+            logger.info("no files modified.")
```
[the remark this answers is not in the document]
Comment (underyx) [reply cli/src/semgrep/autofix.py:258]: /semgrep ignore this is not the app

File: perf/benchmark-against-version.py [hunk 2]
```diff
@@ -68,7 +69,7 @@ def run_semgrep_version(
         logger.info(f"Installing semgrep from {semgrep_local_path.absolute()}")
         subprocess.run(["pip", "install", semgrep_local_path])
     elif semgrep_version == "latest":
-        logger.info(f"Installing latest semgrep")
+        logger.info("Installing latest semgrep")
```
[the remark this answers is not in the document]
Comment (underyx) [reply perf/benchmark-against-version.py:72]: /semgrep ignore this is meant for app only

**Comment by aryx** [comment]
Are those autofixes important? It's just that for osemgrep I've made a snapshot of the semgrep cli code in src/osemgrep/TOPORT/ and gradually ported the code there, and then I was hoping to go through all the changes to the cli that happened since the time I did this snapshot by doing a simple diffs between the date of the snapshot and the current code, but now with all those cosmetic changes, the diff will be very large which is annoying. (I can always split the diff in two, first diff between my snapshot and the code just before this PR, and then another diff after, but it's extra work).

**Comment by aryx** [comment]
Or could you restrict those changes just to the semdep/ directory?

**Comment by underyx** [comment]
@aryx got it, makes sense, I'll close this until I figure out a better plan!

**Pull request closed without merging**